testsuite/tests/c-api/alloc_async.ml

```diff
@@ -1,19 +1,26 @@
 (* TEST
  modules = "alloc_async_stubs.c";
- reason = "alloc async changes: https://github.com/ocaml/ocaml/pull/8897";
- skip;
 *)
 
 external test : int ref -> unit = "stub"
+external print_status : string -> int -> unit = "print_status_caml" [@@noalloc]
+
+(* This tests checks that the finaliser does not run during various
+   allocations from C, but runs at the first polling location in OCaml
+   code after that.
+
+   See in particular RET_FROM_C_CALL from runtime/amd64.S.
```

Comment:
Is `RET_FROM_CALL` _necessary_ for this test to pass? Can you please clarify?

Reply:
Yes, the test now checks that callbacks run at the first polling location in OCaml code, and the test fails by disabling RET_FROM_C_CALL. I will see if I can clarify.

Note that the intention was to include this as part of #11596, but it is possible to merge it with the present PR and close #11596.

Reply:
I wouldn’t mind the text being part of this PR.

Reply:
Ok, it is simpler this way